test/tools/yulInterpreter/Interpreter.cpp

```diff
@@ -150,7 +150,7 @@ void Interpreter::operator()(Block const& _block)
 		if (m_state.loopState != LoopState::Default)
 			break;
 	}
-
+	m_state.loopState = LoopState::Default;
```

Comment:
Hm, I think this is not the right place to reset. Consider a break inside two nested blocks - it should still exit the outer block. My guess would be that we have to reset in the ForLoop just before we enter the post block.

Reply:
```
{
for {} 1 {} {
  for {} 1 {} {
    break
  }
}
```

breaks only from the inner-most for loop, right?

Reply:
Yes, but I was talking about nested blocks, not nested for loops:
`for {} 1 {} { { { { break } } } }`

Reply:
Ah, I think I understand what you are talking about. That reset already exists, the one added was missing.

The reset you are probably referring to is here

https://github.com/ethereum/solidity/blob/cbc1b97760b1ce823962613074239812aba647dd/test/tools/yulInterpreter/Interpreter.cpp#L106-L116

Reply:
sorry, you are right. The fuzzer discovered this

```
{
let a_0, a_1, a_2, a_3
function foo_0(x_0) -> x_1
{
for {} gt(1,x_1) {}
{
if 1 {
break
}
x_1 := 1
}
}
a_0 := foo_0(calldataload(0))
sstore(0, a_0)
}
```

that should store 0 but with this PR stores 1

Reply:
Hmm... I'm beginning to think the problem is for recursive function calls, so perhaps I should be using a stack to store loop state.

Reply:
I'm not sure - a loop state is a very temporary thing

Reply:
Once it is set to anything non-default, the block processor just keeps on returning until a loop is reached.